src/codegen/codegen_llvm.cpp

```diff
@@ -715,7 +1012,19 @@ void CodeGen_LLVM::visit(const GetProperty* op) {
     case TensorProperty::ComponentSize:
     case TensorProperty::ModeOrdering:
     case TensorProperty::ModeTypes:
-    case TensorProperty::Indices:
+    case TensorProperty::Indices: {
+      auto* mode = get_int_constant(32, op->mode, *this->Context);
+      auto* index = get_int_constant(32, op->index, *this->Context);
+
+      auto* load1 = this->Builder->CreateLoad(
+          this->Builder->CreateStructGEP(tensor, (int) TensorProperty::Indices));  // i8***
+      auto* load2 =
+          this->Builder->CreateLoad(this->Builder->CreateInBoundsGEP(load1, mode));  // i8**
+      auto* load3 =
+          this->Builder->CreateLoad(this->Builder->CreateInBoundsGEP(load2, index));  // i8*
+      value = this->Builder->CreateBitCast(load3, i32p);
+      break;
```

Comment:
Let's move this to a function. This might make things easier to debug

Reply:
ok!